### DISCUSSION TITLE
Prediction class - get_train_array(self, j)

### DISCUSSION THREAD

**Discussion by shmh40** [jakobrunge/tigramite#222] [Q&A] [answered]
Hi Jakob and all - firstly, awesome package.

I would like to retrieve the training dataset target values when using the Prediction class. It is easy to retrieve the test set, with pred.get_test_array()[0], as in the tutorial. For the train array, there is a method implemented:

```   
def get_train_array(self, j):
        """Returns training array."""
        return self.fitted_model[j]['data'] 
```

What is j in this context? And what exactly is 'data' referring to? I can't quite work it out at the moment, I'm afraid! 

Thanks!

**Comment by jakobrunge** [comment, marked as answer]
This has been re-worked now. 'j' is the index of the target variable. The array is extracted from the dataframe as desired for the given purpose, this may include masking and data transformation. The array is the result of that.
